### ISSUE TITLE
Reports are lost, depending on order of processing

### ISSUE THREAD

**Issue by mvidner** [YorickPeterse/ruby-lint#174]
```console
$ echo "foo *bar" > a.rb

$ echo "foo"      > b.rb

$ ruby-lint a.rb b.rb
a.rb: warning: line 1, column 5: `*' interpreted as argument prefix

$ ruby-lint b.rb a.rb
a.rb: warning: line 1, column 5: `*' interpreted as argument prefix
b.rb: error: line 1, column 1: undefined method foo
```

**Comment by mvidner** [comment]
An interesting twist is that if both files contain just `foo`, both errors are reported, and if both files contain `foo *bar` then only the first one analyzed is reported.

**Comment by mvidner** [comment]
OK, found the problem: `Runner#analyze` has an explicit FIXME for this: https://github.com/YorickPeterse/ruby-lint/commit/310111a5484a3b4a04e76591f6f830ec0046ba98 (3 years old) citing a bug in Racc.

If I comment out the `return` it does not hang on my test case. Can the workaround be removed now?

**Comment by YorickPeterse** [comment]
@mvidner I'm not sure, I vaguely recall something about that error still being present in the pure Ruby version. At the time JRuby used it but I believe they have since added a JRuby extension for it. If it works on MRI, JRuby, and Rubinius I'm OK with removing the return.

**Comment by mvidner** [comment]
Ah, this is actually a duplicate of #148.
